File: backend/app/schemas/menu.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class MenuItemBase(BaseModel):
    name: str
    description: Optional[str] = None
    price: float
    image_url: Optional[str] = None
    category_id: int
    is_available: bool = True
    stock: int = 0


class MenuItemCreate(MenuItemBase):
    pass


class MenuItemUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    price: Optional[float] = None
    image_url: Optional[str] = None
    category_id: Optional[int] = None
    is_available: Optional[bool] = None
    stock: Optional[int] = None


class MenuItemResponse(MenuItemBase):
    id: int
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/api/menu.ts
import { api } from "./client";

export interface MenuItem {
  id: number;
  name: string;
  description: string | null;
  price: number;
  image_url: string | null;
  category_id: number;
  is_available: boolean;
  stock: number;
  created_at: string;
}

export async function getMenuItems(): Promise<MenuItem[]> {
  const response = await api.get<MenuItem[]>("/menu/");
  return response.data;
}

export async function getMenuItem(
  itemId: number,
): Promise<MenuItem> {
  const response = await api.get<MenuItem>(`/menu/${itemId}`);
  return response.data;
}